src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml.cs

```diff
@@ -128,12 +128,14 @@ internal void DependencyRegen(WorkspaceModel ws)
 
             if (packageDependencies.Any())
             {
+                Boolean hasPackageMarkedForUninstall = false;
                 // If package is set to uninstall state, update the package info
                 foreach (var package in dependencyViewExtension.pmExtension.PackageLoader.LocalPackages.Where(x => x.MarkedForUninstall))
                 {
                     (packageDependencies.Where(x => x.Name == package.Name).FirstOrDefault() as PackageDependencyInfo).State = PackageDependencyState.RequiresRestart;
-                    this.RestartBanner.Visibility = Visibility.Visible;
+                    hasPackageMarkedForUninstall = true;
```

Comment:
@QilongTang sorry, I did not see that line 135 actually sets the state!
I am not sure if this change is actually an optimization - it's likely that the Visibility does not actually trigger any updates unless it changes state.

I am okay if you want to close this.

Reply:
@mjkkirschner If I write this again I may still write it this way, I am not sure about visibility change performance as well